dev/wlp-gradle/java.gradle

```diff
@@ -94,13 +94,18 @@ subprojects {
 
   
   tasks.withType(JavaCompile).configureEach({
+      int javaVersionNumVal = Integer.valueOf(JavaVersion.current().getMajorVersion())
       // javac 12 has a min target of 1.7, so upgrade any 1.6 projects to 1.7 if running on Java 12
-      if (JavaVersion.current().isJava12Compatible()) {
-        if ('1.6'.equals(sourceCompatibility)) {
+      // javac 20 has a min target of 1.8, so upgrade any 1.7 projects to 1.8 if running on Java 20
+      
+      if (('1.6'.equals(sourceCompatibility) || '1.7'.equals(sourceCompatibility)) && javaVersionNumVal >= 20) {
+          sourceCompatibility = "1.8"
+          targetCompatibility = "1.8"
+      } else if ('1.6'.equals(sourceCompatibility) && javaVersionNumVal >= 12) {
           sourceCompatibility = "1.7"
           targetCompatibility = "1.7"
-        }
       }
+      
```

Comment:
```suggestion

```

Reply:
The line has spaces.  Might as well make it an empty line with not spaces.  Not reason to have the line indented.

Reply:
updated